test/gtest/bn_fwd_infer_ocl_hip.cpp

```diff
@@ -0,0 +1,274 @@
+    : public ::testing::TestWithParam<std::tuple<BNTestCase, miopenTensorLayout_t>>
+{
+protected:
+    void SetUp() override
+    {
+        std::tie(bn_config, tensor_layout) = GetParam();
+        bn_infer_test_data.SetUpImpl(bn_config, tensor_layout);
+
+    }
+
+    void RunTestGPU(bool hip_en = false)
+    {
+
+        auto&& handle = get_handle();
+
+        BatchNormForwardInferencGPU(handle,
+                                    bn_config.mode,
+                                    &bn_infer_test_data.alpha,
+                                    &bn_infer_test_data.beta,
+                                    bn_infer_test_data.input.desc,
+                                    bn_infer_test_data.in_dev.get(),
+                                    bn_infer_test_data.output.desc,
+                                    bn_infer_test_data.out_dev.get(),
+                                    bn_infer_test_data.scale.desc,
+                                    bn_infer_test_data.scale_dev.get(),
+                                    bn_infer_test_data.shift_dev.get(),
+                                    bn_infer_test_data.estMean_dev.get(),
+                                    bn_infer_test_data.estVariance_dev.get(),
+                                    bn_infer_test_data.epsilon,
+                                    hip_en);
+
+        auto& output_ref = hip_en ? bn_infer_test_data.output.data : bn_infer_test_data.ref_out.data;
+        
+        std::fill(output_ref.begin(),
+                  output_ref.end(),
+                  std::numeric_limits<YDataType>::quiet_NaN());
+
+
+        output_ref       = handle.Read<YDataType>(bn_infer_test_data.out_dev,
+                                            bn_infer_test_data.output.data.size());
+    }
+
+    void ComputeCPU() { test::ComputeCPUBNInference(bn_infer_test_data); }
+
+    void Verify() { BNTensorCompare<YDataType>(bn_infer_test_data.output, bn_infer_test_data.ref_out, 4e-3); }
+
+
+    BNTestCase bn_config;
+    bool test_skipped = false;
+    BNInferTestData<XDataType, YDataType, ScaleDataType, BiasDataType, MeanVarDataType, BNTestCase>
+        bn_infer_test_data;
+
+    miopenTensorLayout_t tensor_layout;
+};
+
+namespace BatchNormFwdInfer {
+
+struct BatchNormFwdInferTestFloat : BatchNormFwdInferTest<float, float, float, float, float>
+{
+};
+
+} // namespace BatchNormFwdInfer
+using namespace BatchNormFwdInfer;
+
+TEST_P(BatchNormFwdInferTestFloat, BatchNormFwdInferTestFw)
+{
+    // Run the OpenCL reference
+    RunTestGPU(false);
+
+    // Optionally use the CPU output as reference 
+    // ComputeCPU();
+
+    // Run the HIP kernel
+    RunTestGPU(true);
+
+    // Compare the outputs.
+    Verify();
+};
+
+INSTANTIATE_TEST_SUITE_P(BatchNormFwdInferTestSet,
+                         BatchNormFwdInferTestFloat,
+                         testing::Combine(testing::ValuesIn(Network1<BNTestCase>()),
+                                          testing::ValuesIn({miopenTensorNCHW})));
```

Comment:
```suggestion
                                          testing::Values(miopenTensorNCHW)));
```
Probably it's a better way to pass single value, or even multiple ones like `testing::Values(miopenTensorNCHW, miopenTensorNHWC)`.